### DISCUSSION TITLE
Questions about Euphony's functionality

### DISCUSSION THREAD

**Discussion by Kimhaedam** [euphony-io/euphony#143] [Q&A] [answered]
Hi, I have a question about Euphony's function during project idea planning.
My questions are...

- When multiple Android devices send sound waves(not simultaneously), can the listener distinguish between them and identify one device?

- And, can an Android device function both as a speaker and a listener? For example, if the Android phone can function as a speaker and then switch it right away to function as a listener.

I'd appreciate it if you could let me know!

**Comment by yeonns** [comment, marked as answer]
@Kimhaedam Good questions! 👍 

### 1. When multiple Android devices send sound waves(not simultaneously), can the listener distinguish between them and identify one device?
Sometimes it's hard to hear what I'm saying in noisy places. I think it is the same because euphony is also a sound.

### 2. And, can an Android device function both as a speaker and a listener? For example, if the Android phone can function as a speaker and then switch it right away to function as a listener.
That is [euphony-hub-collector](https://github.com/euphony-io/euphony-hub-collector) application. If you run this application in your device, it generates the sound with ```txManager``` and it receives the sound with ```rxManager```. Then, it compares whether the sending result and the receiving result are the same and returns ```success``` or ```fail```.

**Reply by designe** [reply]
For 1st question, 
@yeonns wrote a good example.
Current euphony is a situation where all of them have the same tone of voice. From the listener's point of view, data can be recognized by separating it by frequency. Unlike humans, frequencies can be analyzed separately. Using this to generate different frequency domains, it is possible to recognize data at the same time when data comes out simultaneously.
Currently, there is no implementation. If necessary, the speaker part and the listener part need to be modified.

**Reply by designe** [reply]
I think it is a good question :) if you have an interest about it, we can make a concept `channel` in the euphony.
Do you know WiFi's channel concept? There are 13 channels in 2.4ghz of WiFi.
![image](https://user-images.githubusercontent.com/1748714/178747421-6cd6dcd8-16ed-4d7e-800a-34670645c4dd.png)

But as you know that, euphony is now using only 1 channel :)
So, if it'll be possible, we need to make more chanenls. are you getting me?

**Reply by Kimhaedam** [reply]
Thank you for your kind explanation, It'll be a big help in project planning!
I'll take into consideration that Euphony has one channel. If a channel is added to Euphony like Wi-Fi, I think we can do more things. It's interesting🤩

**Reply by designe** [reply]
exactly :)
